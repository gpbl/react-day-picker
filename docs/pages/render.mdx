import { Render } from '@components/Render';
import { getSrc } from '@utils/getSrc';
import * as examples from '@examples/index';

export const getStaticProps = () =>
  getSrc([
    'ContainerAttributes.tsx',
    'Controlled.tsx',
    'CssVariables.tsx',
    'CustomCaption.tsx',
    'CustomDay.tsx',
    'CustomDisableRow.tsx',
    'CustomMultiple.tsx',
    'CustomSingle.tsx',
    'Popover.tsx',
    'DefaultMonth.tsx',
    'Disabled.tsx',
    'DisableNavigation.tsx',
    'Dropdown.tsx',
    'DropdownButtons.tsx',
    'DropdownMultipleMonths.tsx',
    'Fixedweeks.tsx',
    // 'FocusRecursive.tsx',
    // 'Formatters.tsx',
    // 'FromToMonth.tsx',
    // 'FromToYear.tsx',
    // 'InputRange.tsx',
    // 'InputTime.tsx',
    // 'Keyboard.tsx',
    // 'ModifiersClassnames.tsx',
    // 'ModifiersCustom.tsx',
    // 'ModifiersDisabled.tsx',
    // 'ModifiersHidden.tsx',
    // 'ModifiersSelected.tsx',
    // 'ModifiersStyle.tsx',
    // 'ModifiersToday.tsx',
    // 'Multiple.tsx',
    // 'MultipleMinMax.tsx',
    // 'MultipleMonths.tsx',
    // 'MultipleMonthsId.tsx',
    // 'MultipleMonthsPaged.tsx',
    // 'NumberingSystem.tsx',
    // 'OutsideDays.tsx',
    'Range.tsx',
    'RangeMinMax.tsx',
    'RangeShiftKey.tsx'
    // 'Rtl.tsx',
    // 'Single.tsx',
    // 'SingleRequired.tsx',
    // 'Spanish.tsx',
    // 'SpanishWeekStartsOn.tsx',
    // 'Start.tsx',
    // 'StylingCss.tsx',
    // 'StylingCssModules.tsx',
    // 'StylingInline.tsx',
    // 'StylingModifiers.tsx',
    // 'Testcase1567.tsx',
    // 'WeekIso.tsx',
    // 'Weeknumber.tsx',
    // 'WeeknumberCustom.tsx',
    // 'WeeknumberIso.tsx'
  ]);

<Render />
