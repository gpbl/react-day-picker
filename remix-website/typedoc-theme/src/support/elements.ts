/**
 * A set of pure functions that returns markdown elements as strings.
 *
 * @module
 */

import { formatContents, trimLastLine, unEscapeChars } from './utils.js';

export const heading = (level: number, text: string) => {
  level = level > 6 ? 6 : level;
  return `${[...Array(level)].map(() => '#').join('')} ${text}`;
};

export const link = (label: string, url: string | null) =>
  url ? `[${label}](${url})` : '';

export const bold = (text: string) => `**${text}**`;

export const italic = (text: string) => `*${text}*`;

export const backTicks = (text: string) =>
  /(\`)/g.test(text) ? text.replace(/`/g, '\\`') : `\`${text}\``;

export const unorderedList = <T>(items: T[]) =>
  items.map((item) => `- ${item}`).join('\n');

export const horizontalRule = () => '\n\n***\n\n';

export const codeBlock = (content: string) => {
  const trimmedContent =
    content.endsWith('}') ||
    content.endsWith('};') ||
    content.endsWith('>') ||
    content.endsWith('>;')
      ? trimLastLine(content)
      : content;
  return '```ts\n' + unEscapeChars(trimmedContent) + '\n```';
};

export const strikeThrough = (content: string) => `~~${content}~~`;

export const table = (headers: string[], rows: string[][]) =>
  `\n| ${headers.join(' | ')} |\n| ${headers
    .map(() => ':------')
    .join(' | ')} |\n${rows.map((row) => `| ${row.join(' | ')} |\n`).join('')}`;

export const blockQuoteBlock = (content: string) => {
  const lines = formatContents(content).split('\n');
  return lines
    .map((line) => (line.length ? `> ${line.trim()}` : '>'))
    .join('\n');
};

export const indentBlock = (content: string) => {
  const lines = content.split('\n');
  return lines
    .filter((line) => Boolean(line.length))
    .map((line) => `    ${line}`)
    .join('\n');
};
