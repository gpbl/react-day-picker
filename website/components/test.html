<!DOCTYPE html>
<html>
<head>
<style>
body {
  display: flex;
  margin: 0;
  padding: 0;
  min-height: 100vh;
  overflow-x: hidden;
}

.sidebar {
  position: sticky;
  top: 0;
  height: 100vh;
  width: 200px;
  background-color: #f5f5f5;
  padding: 20px;
  overflow-y: auto;
}

.main-content {
  flex-grow: 1;
  padding: 20px;
}
</style>
</head>
<body>

<div class="sidebar">
  <h2>Sidebar</h2>
  <p>This is a sticky sidebar.</p>
</div>

<div class="main-content">
  <h2>Main Content</h2>
  <p>Scroll the page.</p>
  <!-- Add enough content here so that the page scrolls -->
</div>

</body>
</html>
