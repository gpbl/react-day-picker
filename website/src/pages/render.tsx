/* eslint-disable @typescript-eslint/no-var-requires */
import React from 'react';
import { useLocation } from '@docusaurus/router';
import BrowserOnly from '@docusaurus/BrowserOnly';

/**
 * Create a page at `/render` URL that renders the examples from the examples
 * `directory`.
 * */
export default function Render(): JSX.Element {
  const location = useLocation();
  return (
    <BrowserOnly>
      {() => {
        const fileName = new URLSearchParams(location.search).get('example');
        try {
          require(`../../examples/${fileName}`).default;
        } catch (e) {
          console.error('Error requiring %s', `../../examples/${fileName}`, e);
          return <pre>{e.message}</pre>;
        }
        const Component = require(`../../examples/${fileName}`).default;
        return <Component />;
      }}
    </BrowserOnly>
  );
}
