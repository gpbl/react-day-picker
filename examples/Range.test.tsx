import React from "react";

import { addDays } from "date-fns";

import { dateButton, gridcell } from "@/test/elements";
import { render, screen } from "@/test/render";
import { user } from "@/test/user";

import { Range } from "./Range";

const defaultMonth = new Date(2020, 5, 15);

let container: HTMLElement;
const days = [
  defaultMonth,
  addDays(defaultMonth, 1),
  addDays(defaultMonth, 2),
  addDays(defaultMonth, 3),
  addDays(defaultMonth, 4)
];

beforeEach(() => (container = render(<Range />).container));

test("should match the snapshot", () => {
  expect(container).toMatchSnapshot();
});
test.each(days)("%s should be selected", (day) => {
  expect(gridcell(day, true)).toHaveAttribute("aria-selected", "true");
});

describe("when a day in the range is clicked", () => {
  const day = days[2];
  beforeEach(async () => user.click(dateButton(day)));
  test.each([days[0], days[1], day])("%s should be selected", (day) => {
    expect(gridcell(day, true)).toHaveAttribute("aria-selected", "true");
  });
  test.each([days[3], days[4]])("%s should not be selected", (day) => {
    expect(gridcell(day, true)).not.toHaveAttribute("aria-selected");
  });
  describe("when the day is clicked again", () => {
    const day = days[2];
    beforeEach(async () => user.click(dateButton(day)));
    test("only one day should be selected", () => {
      expect(getAllSelected()).toHaveLength(1);
    });
    test("only a day in the range should be selected", () => {
      expect(gridcell(day, true)).toHaveAttribute("aria-selected", "true");
    });

    describe("when a day in the range is clicked again", () => {
      const day = days[2];
      beforeEach(async () => user.click(dateButton(day)));
      test("no day should be selected", () => {
        expect(getAllSelected()).toHaveLength(0);
      });
      test("should match the snapshot", () => {
        expect(container).toMatchSnapshot();
      });
    });
  });
});

function getAllSelected() {
  const gridcells = screen.getAllByRole("gridcell");

  return Array.from(gridcells).filter(
    (gridcell) => gridcell.getAttribute("aria-selected") === "true"
  );
}
