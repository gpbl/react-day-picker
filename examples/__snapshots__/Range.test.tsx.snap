// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="rdp-root"
      data-mode="range"
      id="test"
    >
      <div
        class="rdp-months"
      >
        <nav
          aria-label=""
          class="rdp-nav"
        >
          <button
            aria-label="Go to the Previous Month"
            class="rdp-button_previous"
            type="button"
          >
            <svg
              class="rdp-chevron"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <polygon
                points="16 18.112 9.81111111 12 16 5.87733333 14.0888889 4 6 12 14.0888889 20"
              />
            </svg>
          </button>
          <button
            aria-label="Go to the Next Month"
            class="rdp-button_next"
            type="button"
          >
            <svg
              class="rdp-chevron"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <polygon
                points="8 18.612 14.1888889 12.5 8 6.37733333 9.91111111 4.5 18 12.5 9.91111111 20.5"
              />
            </svg>
          </button>
        </nav>
        <div
          class="rdp-month"
        >
          <div
            class="rdp-month_caption"
          >
            <span
              aria-live="polite"
              class="rdp-caption_label"
              role="status"
            >
              June 2020
            </span>
          </div>
          <table
            aria-label="June 2020"
            aria-multiselectable="true"
            class="rdp-month_grid"
            role="grid"
          >
            <thead
              aria-hidden="true"
            >
              <tr
                class="rdp-weekdays"
              >
                <th
                  aria-label="Sunday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Su
                </th>
                <th
                  aria-label="Monday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Mo
                </th>
                <th
                  aria-label="Tuesday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Tu
                </th>
                <th
                  aria-label="Wednesday"
                  class="rdp-weekday"
                  scope="col"
                >
                  We
                </th>
                <th
                  aria-label="Thursday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Th
                </th>
                <th
                  aria-label="Friday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Fr
                </th>
                <th
                  aria-label="Saturday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Sa
                </th>
              </tr>
            </thead>
            <tbody
              class="rdp-weeks"
            >
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-05-31"
                  data-hidden="true"
                  data-month="2020-05"
                  data-outside="true"
                  role="gridcell"
                />
                <td
                  class="rdp-day"
                  data-day="2020-06-01"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 1st, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    1
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-02"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 2nd, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    2
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-03"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, June 3rd, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    3
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-04"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, June 4th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    4
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-05"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, June 5th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    5
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-06"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, June 6th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    6
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2020-06-07"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, June 7th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    7
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-08"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 8th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    8
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-09"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 9th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    9
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-10"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, June 10th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    10
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-11"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, June 11th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    11
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-12"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, June 12th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    12
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-13"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, June 13th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    13
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2020-06-14"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, June 14th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    14
                  </button>
                </td>
                <td
                  aria-selected="true"
                  class="rdp-day rdp-selected rdp-range_start"
                  data-day="2020-06-15"
                  data-selected="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 15th, 2020, selected"
                    class="rdp-day_button"
                    tabindex="0"
                    type="button"
                  >
                    15
                  </button>
                </td>
                <td
                  aria-selected="true"
                  class="rdp-day rdp-selected rdp-range_middle"
                  data-day="2020-06-16"
                  data-selected="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 16th, 2020, selected"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    16
                  </button>
                </td>
                <td
                  aria-selected="true"
                  class="rdp-day rdp-selected rdp-range_middle"
                  data-day="2020-06-17"
                  data-selected="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, June 17th, 2020, selected"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    17
                  </button>
                </td>
                <td
                  aria-selected="true"
                  class="rdp-day rdp-selected rdp-range_middle"
                  data-day="2020-06-18"
                  data-selected="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, June 18th, 2020, selected"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    18
                  </button>
                </td>
                <td
                  aria-selected="true"
                  class="rdp-day rdp-selected rdp-range_end"
                  data-day="2020-06-19"
                  data-selected="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, June 19th, 2020, selected"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    19
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-20"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, June 20th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    20
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2020-06-21"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, June 21st, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    21
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-22"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 22nd, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    22
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-23"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 23rd, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    23
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-24"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, June 24th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    24
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-25"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, June 25th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    25
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-26"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, June 26th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    26
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-27"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, June 27th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    27
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2020-06-28"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, June 28th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    28
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-29"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 29th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    29
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-30"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 30th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    30
                  </button>
                </td>
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-07-01"
                  data-hidden="true"
                  data-month="2020-07"
                  data-outside="true"
                  role="gridcell"
                />
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-07-02"
                  data-hidden="true"
                  data-month="2020-07"
                  data-outside="true"
                  role="gridcell"
                />
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-07-03"
                  data-hidden="true"
                  data-month="2020-07"
                  data-outside="true"
                  role="gridcell"
                />
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-07-04"
                  data-hidden="true"
                  data-month="2020-07"
                  data-outside="true"
                  role="gridcell"
                />
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        aria-live="polite"
        class="rdp-footer"
        role="status"
      >
        June 15th, 2020–June 19th, 2020
      </div>
    </div>
    <button
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;

exports[`when a day in the range is clicked when the day is clicked again when a day in the range is clicked again should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="rdp-root"
      data-mode="range"
      id="test"
    >
      <div
        class="rdp-months"
      >
        <nav
          aria-label=""
          class="rdp-nav"
        >
          <button
            aria-label="Go to the Previous Month"
            class="rdp-button_previous"
            type="button"
          >
            <svg
              class="rdp-chevron"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <polygon
                points="16 18.112 9.81111111 12 16 5.87733333 14.0888889 4 6 12 14.0888889 20"
              />
            </svg>
          </button>
          <button
            aria-label="Go to the Next Month"
            class="rdp-button_next"
            type="button"
          >
            <svg
              class="rdp-chevron"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <polygon
                points="8 18.612 14.1888889 12.5 8 6.37733333 9.91111111 4.5 18 12.5 9.91111111 20.5"
              />
            </svg>
          </button>
        </nav>
        <div
          class="rdp-month"
        >
          <div
            class="rdp-month_caption"
          >
            <span
              aria-live="polite"
              class="rdp-caption_label"
              role="status"
            >
              June 2020
            </span>
          </div>
          <table
            aria-label="June 2020"
            aria-multiselectable="true"
            class="rdp-month_grid"
            role="grid"
          >
            <thead
              aria-hidden="true"
            >
              <tr
                class="rdp-weekdays"
              >
                <th
                  aria-label="Sunday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Su
                </th>
                <th
                  aria-label="Monday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Mo
                </th>
                <th
                  aria-label="Tuesday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Tu
                </th>
                <th
                  aria-label="Wednesday"
                  class="rdp-weekday"
                  scope="col"
                >
                  We
                </th>
                <th
                  aria-label="Thursday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Th
                </th>
                <th
                  aria-label="Friday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Fr
                </th>
                <th
                  aria-label="Saturday"
                  class="rdp-weekday"
                  scope="col"
                >
                  Sa
                </th>
              </tr>
            </thead>
            <tbody
              class="rdp-weeks"
            >
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-05-31"
                  data-hidden="true"
                  data-month="2020-05"
                  data-outside="true"
                  role="gridcell"
                />
                <td
                  class="rdp-day"
                  data-day="2020-06-01"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 1st, 2020"
                    class="rdp-day_button"
                    tabindex="0"
                    type="button"
                  >
                    1
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-02"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 2nd, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    2
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-03"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, June 3rd, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    3
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-04"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, June 4th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    4
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-05"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, June 5th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    5
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-06"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, June 6th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    6
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2020-06-07"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, June 7th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    7
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-08"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 8th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    8
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-09"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 9th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    9
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-10"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, June 10th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    10
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-11"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, June 11th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    11
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-12"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, June 12th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    12
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-13"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, June 13th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    13
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2020-06-14"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, June 14th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    14
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-15"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 15th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    15
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-16"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 16th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    16
                  </button>
                </td>
                <td
                  class="rdp-day rdp-focused"
                  data-day="2020-06-17"
                  data-focused="true"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, June 17th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    17
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-18"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, June 18th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    18
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-19"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, June 19th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    19
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-20"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, June 20th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    20
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2020-06-21"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, June 21st, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    21
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-22"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 22nd, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    22
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-23"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 23rd, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    23
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-24"
                  role="gridcell"
                >
                  <button
                    aria-label="Wednesday, June 24th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    24
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-25"
                  role="gridcell"
                >
                  <button
                    aria-label="Thursday, June 25th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    25
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-26"
                  role="gridcell"
                >
                  <button
                    aria-label="Friday, June 26th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    26
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-27"
                  role="gridcell"
                >
                  <button
                    aria-label="Saturday, June 27th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    27
                  </button>
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  class="rdp-day"
                  data-day="2020-06-28"
                  role="gridcell"
                >
                  <button
                    aria-label="Sunday, June 28th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    28
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-29"
                  role="gridcell"
                >
                  <button
                    aria-label="Monday, June 29th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    29
                  </button>
                </td>
                <td
                  class="rdp-day"
                  data-day="2020-06-30"
                  role="gridcell"
                >
                  <button
                    aria-label="Tuesday, June 30th, 2020"
                    class="rdp-day_button"
                    tabindex="-1"
                    type="button"
                  >
                    30
                  </button>
                </td>
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-07-01"
                  data-hidden="true"
                  data-month="2020-07"
                  data-outside="true"
                  role="gridcell"
                />
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-07-02"
                  data-hidden="true"
                  data-month="2020-07"
                  data-outside="true"
                  role="gridcell"
                />
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-07-03"
                  data-hidden="true"
                  data-month="2020-07"
                  data-outside="true"
                  role="gridcell"
                />
                <td
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2020-07-04"
                  data-hidden="true"
                  data-month="2020-07"
                  data-outside="true"
                  role="gridcell"
                />
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        aria-live="polite"
        class="rdp-footer"
        role="status"
      >
        Please pick the first day.
      </div>
    </div>
    <button
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;
